### PR TITLE
[ETHOSN] Implement tanh operator

### PR DESCRIPTION
**Pull request by Leo-arm**
Adding compiler support for TANH operator, which is based on
an underlying pattern matching scheme.

One negative test is included as well.

Co-authored-by: Samuel Panijel <samuel.panijel@arm.com>

### CODE REVIEW

File: tests/python/contrib/test_ethosn/test_tanh.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        tanh,
+        output_scale=relay.const(output_sc, "float32"),
+        output_zero_point=relay.const(output_zp, "int32"),
+        out_dtype=dtype,
+    )
+    return model
+
+
+@requires_ethosn
+@pytest.mark.parametrize("shape", [(1, 512, 512, 3)])
```
Comment (manupa-arm) [inline tests/python/contrib/test_ethosn/test_tanh.py:46]: Should we not need more test cases for shapes, scales and zero point ?

Happy to take on in a follow up.

**Review by manupa-arm — COMMENTED** [review]
Just a minor comment (can take in a follow up...)

File: tests/python/contrib/test_ethosn/test_tanh.py (hunk 1 above)
Comment (Leo-arm) [reply tests/python/contrib/test_ethosn/test_tanh.py:46]: The variability is really taken care of in the Ethos-N driver stack which is tested separately; the TVM integration code would be the same for the other test cases. A follow-on PR will add int8 support and reduce the size of the tensor as this is a little excessive, for no good reason.

Comment (manupa-arm) [reply tests/python/contrib/test_ethosn/test_tanh.py:46]: Ack

**Review by manupa-arm — APPROVED** [review]
LGTM!

**Comment by manupa-arm** [comment]
Thanks @Leo-arm !
